e2e/Microsoft.Azure.Devices.E2ETests/TwinE2ETests.cs

```diff
@@ -661,5 +728,84 @@ static public void ClassCleanup()
             TestContext.WriteLine("verified " + serviceTwin.Properties.Reported[propName].ToString() + "=" + propValue);
             await TestUtil.RemoveDeviceAsync(deviceInfo.Item1, registryManager);
         }
+
+        private async Task _Twin_ServiceDoesNotCreateNullPropertyInCollection(Client.TransportType transport)
+        {
+            var propName1 = Guid.NewGuid().ToString();
+            var propName2 = Guid.NewGuid().ToString();
+            var propEmptyValue = "{}";
+
+            Tuple<string, string> deviceInfo = TestUtil.CreateDevice(DevicePrefix, hostName, registryManager);
+            var deviceClient = DeviceClient.CreateFromConnectionString(deviceInfo.Item2, transport);
+
+            await deviceClient.UpdateReportedPropertiesAsync(new TwinCollection
+            {
+                [propName1] = null
+            });
+            var serviceTwin = await registryManager.GetTwinAsync(deviceInfo.Item1);
+            Assert.IsFalse(serviceTwin.Properties.Reported.Contains(propName1));
+
+            await deviceClient.UpdateReportedPropertiesAsync(new TwinCollection
+            {
+                [propName1] = new TwinCollection
+                {
+                    [propName2] = null
+                }
+            });
+            serviceTwin = await registryManager.GetTwinAsync(deviceInfo.Item1);
+            Assert.IsTrue(serviceTwin.Properties.Reported.Contains(propName1));
+            String value1 = serviceTwin.Properties.Reported[propName1].ToString();
+
+            // If service team fixed the Issue #243 the following Assert.AreNotEqual should fail 
```

Comment:
Nit: I would start with `// TODO: #243 ` to make work left for the future easy to search.